tensorboard/webapp/widgets/data_table/data_table_test.ts

```diff
@@ -456,4 +506,225 @@ describe('data table', () => {
+      const fixture = createComponent({
+        headers: mockHeaders,
+        data: mockTableData,
+        selectabelColumns: mockSelectableColumns,
+      });
+      const cell = fixture.debugElement.query(
+        By.directive(ContentCellComponent)
+      );
+      cell.nativeElement.dispatchEvent(new MouseEvent('contextmenu'));
+      fixture.detectChanges();
+
+      fixture.debugElement
+        .queryAll(By.css('.context-menu button'))
+        .find((element) => element.nativeElement.innerHTML.includes('Right'))!
+        .nativeElement.click();
+      fixture.detectChanges();
+
+      expect(
+        fixture.debugElement.query(By.directive(ColumnSelectorComponent))
+      ).not.toBeNull();
+    });
+
+    it('only shows the disabled button when the column is hparam', () => {
+      const fixture = createComponent({
+        headers: mockHeaders,
+        data: mockTableData,
+        selectabelColumns: mockSelectableColumns,
+      });
+      const cell = fixture.debugElement.query(
+        By.directive(ContentCellComponent)
+      );
+      cell.nativeElement.dispatchEvent(new MouseEvent('contextmenu'));
+      fixture.detectChanges();
+
+      const disableBtn = fixture.debugElement
+        .queryAll(By.css('.context-menu button'))
+        .find((btn) => btn.nativeElement.innerHTML.includes('Disable'))!;
+
+      expect(disableBtn).toBeUndefined();
+    });
+
+    it('removes column when Disable button is clicked', fakeAsync(() => {
+      const fixture = createComponent({
+        headers: mockHeaders,
+        data: mockTableData,
+        selectabelColumns: mockSelectableColumns,
+      });
+      const cell = fixture.debugElement
+        .queryAll(By.directive(ContentCellComponent))
+        .find((cell) => cell.nativeElement.innerHTML.includes('other header'))!;
+      cell.nativeElement.dispatchEvent(new MouseEvent('contextmenu'));
+      fixture.detectChanges();
+
+      let columnRemoved: ColumnHeader | undefined = undefined;
+      fixture.componentInstance.removeColumn.subscribe((event) => {
+        columnRemoved = event;
+      });
+
+      fixture.debugElement
+        .queryAll(By.css('.context-menu button'))
+        .find((btn) => btn.nativeElement.innerHTML.includes('Remove'))!
+        .nativeElement.click();
+      flush();
+
+      expect(columnRemoved).toBeDefined();
+    }));
+
+    it('does not include add buttons when there are no selectable columns', () => {
+      const fixture = createComponent({
+        headers: mockHeaders,
+        data: mockTableData,
+      });
+      const cell = fixture.debugElement
+        .queryAll(By.directive(ContentCellComponent))
+        .find((cell) => cell.nativeElement.innerHTML.includes('other header'))!;
+      cell.nativeElement.dispatchEvent(new MouseEvent('contextmenu'));
+      fixture.detectChanges();
+
+      expect(
+        fixture.debugElement
+          .queryAll(By.css('.context-menu button'))
```

Comment:
As we discussed offline. There should be a message if no menu buttons appear. Also a test for that message.

Reply:
![image](https://github.com/tensorflow/tensorboard/assets/78179109/2f7e9e4f-0007-49a6-9b88-30f3e5649137)